src/__tests__/components/molecules/LoginStatus.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { mockDeep } from 'jest-mock-extended';
+import { render, waitFor } from '@testing-library/react';
+import { LoginStatus } from '@molecules/LoginStatus/LoginStatus';
+import { Authenticator, useAuthenticator } from '@aws-amplify/ui-react';
+import { fetchAuthSession } from '@aws-amplify/auth';
+
+jest.mock('@aws-amplify/ui-react', () => ({
+  ...jest.requireActual('@aws-amplify/ui-react'),
+  useAuthenticator: jest.fn(),
+}));
+jest.mock('@aws-amplify/auth', () => ({
+  ...jest.requireActual('@aws-amplify/auth'),
+  fetchAuthSession: jest.fn(),
+}));
+
+test('LoginStatus - unauthenticated', async () => {
+  jest.mocked(useAuthenticator).mockReturnValue(
+    mockDeep<ReturnType<typeof useAuthenticator>>({
+      authStatus: 'unauthenticated',
+    })
+  );
+  jest.mocked(fetchAuthSession).mockResolvedValue({
```

Comment:
Why not declare this mock once?

```
const fetchAuthSessionMock = jest.mocked(fetchAuthSession)
```

Also seems like good practice to call `after/beforeEach(jest.resetAllMocks)` to ensure mocks are cleared.

Reply:
updated